webviews/src/views/RokuAutomationView/AutoRunsEditor.svelte

```diff
@@ -0,0 +1,395 @@
+<script>
+    import { Trash, Copy, TriangleDown } from 'svelte-codicons';
+    import { dropzone, draggable } from './dnd';
+
+    export let runs;
+    export let selectedRun;
+
+    let runTable;
+    let runNameInput;
+    let runNameDialog;
+    let runNameDialogValue;
+    let confirmDialog;
+    let showContent = false;
+
+    const toggleDropDown = () => {
+        showContent = !showContent;
+    };
+
+    const selectRun = (e) => {
+        const run = getRunFromEvent(e);
+        selectedRun = run;
+    };
+
+    const moveSelection = (e) => {
+        if (["ArrowUp", "ArrowDown"].includes(e.key)) {
+            const index = runs.findIndex((r) => r.name === selectedRun) + (e.key === "ArrowUp" ? -1 : 1);
+            selectedRun = runs[(index + runs.length) % runs.length].name;
+        }
+    };
+
+    const copyRun = (e) => {
```

Comment:
Convert this to an async function that leverages `await` instead of using promises. (code is cleaner that way)

Reply:
yep